catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/MetacardImpl.java

```diff
@@ -110,7 +113,7 @@ public MetacardImpl(MetacardType type) {
          * serialized object is maintained. For instance, if a null check is added in the
          * constructor, the same check should be added in the readObject() method.
          */
-        map = new HashMap<String, Attribute>();
+        map = new ConcurrentHashMap<>();
```

Comment:
Except that is in the constructor, so the underlying map that is being used to build this metacard is (maybe?) the one being changed? If the calling code (like phil pointed out) is explicitly running two things concurrently on the metacard, it would just be safer to have the copy happen in the cachingfederation strategy. That way you're purposely creating a safe copy for this concurrent process.

<sidenote> 
_**Sigh**_, this is fueling my desire to do a weekend project and make an `ImmutableMetacardImpl`.. that is actually immutable but still has our wrapping semantics.
</sidenote>